test/e2e/managedcluster_test.go

```diff
@@ -25,5 +31,129 @@ var _ = ginkgo.Describe("Testing ManagedCluster", func() {
 				return nil
 			}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
 		})
+
+		ginkgo.It("Get CA from apiserver", func() {
+			//Only need to test this case in ocp
+			if !isOcp {
+				return
+			}
+			//Create a fake secret for apiserver
+			fakesecretName := "fake-server-secret"
+			fakeSecret, err := e2eutil.CreateFakeTlsSecret(kubeClient, fakesecretName, utils.OpenshiftConfigNamespace)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			//get apiserveraddress
+			apiserverAddress, err := utils.GetKubeAPIServerAddress(context.TODO(), ocpClient)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			//add serving secret in apiserver
+			url, err := url.Parse(apiserverAddress)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			apiserver, err := ocpClient.ConfigV1().APIServers().Get(context.TODO(), utils.ApiserverConfigName, metav1.GetOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			newApiserver := apiserver.DeepCopy()
+			newApiserver.Spec.ServingCerts.NamedCertificates = []configv1.APIServerNamedServingCert{
+				{
+					Names: []string{
+						url.Hostname(),
+					},
+					ServingCertificate: configv1.SecretNameReference{
+						Name: fakesecretName,
+					},
+				},
+			}
+
+			newApiserver, err = ocpClient.ConfigV1().APIServers().Update(context.TODO(), newApiserver, metav1.UpdateOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+			gomega.Eventually(func() bool {
+				cluster, err := clusterClient.ClusterV1().ManagedClusters().Get(context.Background(), defaultManagedCluster, metav1.GetOptions{})
+				gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+				if len(cluster.Spec.ManagedClusterClientConfigs) == 0 {
+					return false
+				}
+				for _, config := range cluster.Spec.ManagedClusterClientConfigs {
+					if config.URL != apiserverAddress {
+						continue
+					}
+					if reflect.DeepEqual(config.CABundle, fakeSecret.Data["tls.crt"]) {
+						return true
+					}
+				}
+				return false
+			}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
+
+			//rollback apiserver and delete secret
+			newApiserver.Spec.ServingCerts.NamedCertificates = []configv1.APIServerNamedServingCert{}
+			_, err = ocpClient.ConfigV1().APIServers().Update(context.TODO(), newApiserver, metav1.UpdateOptions{})
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			err = kubeClient.CoreV1().Secrets(utils.OpenshiftConfigNamespace).Delete(context.TODO(), fakesecretName, metav1.DeleteOptions{})
```

Comment:
how about put create fake resource in Before , delete them in After ?

Reply:
I think before and after will run in every case, the current resources are only used in this case